functions/nodejs-apps.bash

```diff
@@ -202,3 +202,147 @@ nodered_setup() {
     dashboard_add_tile "nodered"
   fi
 }
+
+## Function for downloading zigbee2mqtt to current system
+##
+##    zigbee2mqtt_download(String prefix)
+##
+zigbee2mqtt_download() {
+  echo -n "$(timestamp) [openHABian] Downloading zigbee2mqtt... "
+  #if ! [[ -d "${1}/zigbee2mqtt" ]]; then
+    cond_echo "\\nFresh Installation... "
+    if ! cond_redirect mkdir /opt/zigbee2mqtt; then echo "FAILED (mkdir /opt/zigbee2mqtt)";echo1; fi
+    if ! cond_redirect chown openhabian /opt/zigbee2mqtt; then echo "FAILED (chown /opt/zigbee2mqtt)";echo 2; fi
+    if ! cond_redirect chgrp openhab /opt/zigbee2mqtt; then echo "FAILED (chgrp /opt/zigbee2mqtt)";echo 3; fi
+    if ! cond_redirect sudo -u "${username:-openhabian}" git clone https://github.com/Koenkk/zigbee2mqtt.git "/opt/zigbee2mqtt"; then echo "FAILED (git clone)"; return 1; fi
+  #fi
+}
+
+## Function for installing zigbee2mqtt.
+##
+##    zigbee2mqtt_setup()
+##
+zigbee2mqtt_setup() {
+  local zigbee2mqttBase
+  local z2mInstalledText="A configuration for Zigbee2MQTT is already existing.\\n\\nWould you like to update Zigbee2MQTT to the latest version with this configuration?"
+  local introText="A MQTT-server is required for zigbee2mqtt. If you haven't installed one yet, please select <cancel> and come back after installing one (e.g. Mosquitto).\\n\\nZigbee2MQTT will be installed from the official repository.\\n\\nDuration is about 4 minutes..."
+  local installText="Zigbee2MQTT is installed from the official repository.\\n\\nPlease wait about 4 minutes..."
+  local uninstallText="Zigbee2MQTT is completely removed from the system."
+  local adapterText="Please select your zigbee adapter:"
+  local mqttUserText="\\nPlease enter your MQTT-User (default = openhabian):"
+  local mqttPWText="\\nIf your MQTT-server requires a password, please enter it here:"
+  local adapterArray=()
+  local my_adapters=""
+  local mqttDefaultUser="openhabian"
+  local mqttUser
+  local serverIP; serverIP="$(hostname -I)"; serverIP=${serverIP::-1} # remove trailing space
```

Comment:
Assign the variable after all the variables have been declared to keep the style consistent with the rest of the code.

Reply:
yeah, you got me. I needed the hostname for the success text, that's why I squashed it in. I changed that.
Now I separated declaration and initialisation

Reply:
I think you forgot to commit/push the changes.

Reply:
Yes not committed yet. I still have to test but first have to work. 
Many thanks for the helpful hints. I try to finish the corrections this evening.